core/src/main/java/org/apache/iceberg/io/OutputFileFactory.java

```diff
@@ -59,6 +60,11 @@ public OutputFileFactory(PartitionSpec spec, FileFormat format, LocationProvider
     this(spec, format, locations, io, encryptionManager, partitionId, taskId, UUID.randomUUID().toString());
   }
 
+  public OutputFileFactory(Table table, FileFormat format, int partitionId, long taskId) {
```

Comment:
This could be useful here as well:
https://github.com/apache/iceberg/blob/05d39c5dd09d604336ab21c34e78553e5a6b13ef/mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputFormat.java#L76-L81

Reply:
This particular call also passes partition id. I think `OutputFileFactory` should have a builder. I added a TODO item and would propose to refactor this part once we switch to the builder approach. How does that sound, @pvary?

Reply:
I'll resolve this for now.